### DISCUSSION TITLE
`HEARTBEAT_DEAD` for license that doesn't require heartbeat (anymore)

### DISCUSSION THREAD

**Discussion by nschloe** [keygen-sh/community#140] [Issue]
I recently switched a policy from requiring a heartbeat to not requiring one. Now I'm getting `HEARTBEAT_DEAD` error codes for respective licenses. Is this a mistake?

_Edit:_ I checked and licenses do have `requireHeartbeat: False`, but activated machines _do_ have a heartbeat of 600 seconds. After that time, machines are `DEAD` and removed.

_Edit 2:_ I tried it with a pristine policy, everything set to `requireHeartbeat: False`, but machines _still_ require a heartbeat. In fact, I don't know how to create machines without.

**Comment by ezekg** [comment]
Are the machines sending a heartbeat? Looking at your account, that looks like what's happening. If a machine sends a heartbeat, their heartbeat monitor will always be active, even if the policy doesn't necessarily require it. And if they fail to send subsequent heartbeats then they will be culled. So if you don't want machines to be culled, don't require or send a heartbeat and the monitor won't start.

**Reply by ezekg** [reply]
To add — if you want to change a machine that already has an active heartbeat to not require a heartbeat, you must also [reset its heartbeat](https://keygen.sh/docs/api/machines/#machines-actions-reset), otherwise it will still have a heartbeat monitor active, and will be culled after it "dies."

**Reply by nschloe** [reply]
I didn't explicitly send a heartbeat, but I had `Heartbeat Basis` set to `From Creation`. Setting it to `From First Ping` solved the problem.

**Reply by ezekg** [reply]
Ah, well that's not a pristine policy then. The `FROM_CREATION` heartbeat basis is only the default when the policy requires a heartbeat. 🙂
